pkg/collector/corechecks/system/gpu/nvidia/jetson.go

```diff
@@ -0,0 +1,490 @@
+		cpuTags := []string{fmt.Sprintf("cpu:%d", i)}
+		if err != nil {
+			return err
+		}
+		sender.Gauge("nvidia.jetson.cpu.usage", cpuUsage, "", cpuTags)
+
+		cpuFreq, err := strconv.ParseFloat(cpuAndFreqFields[0][cpuFreq], 64)
+		if err != nil {
+			return err
+		}
+		sender.Gauge("nvidia.jetson.cpu.freq", cpuFreq, "", cpuTags)
+	}
+
+	return nil
+}
+
+func (c *JetsonCheck) sendTemperatureMetrics(sender aggregator.Sender, field string) error {
+	temperatureFields := c.regexes[regexTemperatureIdx].FindAllStringSubmatch(field, -1)
+	if len(temperatureFields) <= 0 {
+		return errors.New("could not parse temperature fields")
+	}
+
+	for i := 0; i < len(temperatureFields); i++ {
+		tempValue, err := strconv.ParseFloat(temperatureFields[i][tempValue], 64)
+		if err != nil {
+			return err
+		}
+		temperatureZoneTags := []string{fmt.Sprintf("zone:%s", temperatureFields[i][tempZone])}
+		sender.Gauge("nvidia.jetson.gpu.temp", tempValue, "", temperatureZoneTags)
+	}
+
+	return nil
+}
+
+func (c *JetsonCheck) sendVoltageMetrics(sender aggregator.Sender, field string) error {
+	voltageFields := c.regexes[regexVoltageIdx].FindAllStringSubmatch(field, -1)
+	if len(voltageFields) <= 0 {
+		return errors.New("could not parse voltage fields")
+	}
+
+	for i := 0; i < len(voltageFields); i++ {
+		voltageProbeTags := []string{fmt.Sprintf("probe:%s", voltageFields[i][voltageProbeName])}
+		instantVoltage, err := strconv.ParseFloat(voltageFields[i][currentVoltage], 64)
+		if err != nil {
+			return err
+		}
+		sender.Gauge("nvidia.jetson.gpu.vdd.instant", instantVoltage, "", voltageProbeTags)
+
+		averageVoltage, err := strconv.ParseFloat(voltageFields[i][averageVoltage], 64)
+		if err != nil {
+			return err
+		}
+		sender.Gauge("nvidia.jetson.gpu.vdd.average", averageVoltage, "", voltageProbeTags)
+	}
+
+	return nil
+}
+
+func (c *JetsonCheck) processTegraStatsOutput(tegraStatsOuptut string) error {
+	sender, err := aggregator.GetSender(c.ID())
+	if err != nil {
+		return err
+	}
+
+	var metricSenders = []sendMetric{
+		c.sendRAMMetrics,
+		c.sendIRAMMetrics,
+		c.sendSwapMetrics,
+		c.sendGpuUsageMetrics,
+		c.sendCPUUsageMetrics,
+		c.sendTemperatureMetrics,
+		c.sendVoltageMetrics,
+	}
+	for _, metricSender := range metricSenders {
+		err = metricSender(sender, tegraStatsOuptut)
+		if err != nil {
+			return err
+		}
+	}
+	sender.Commit()
+	return nil
+}
+
+// Run executes the check
+func (c *JetsonCheck) Run() error {
+	cmd := exec.Command(c.tegraStatsPath, c.commandOpts...)
+
+	// Parse the standard output for the stats
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stdout)
+		if in.Scan() {
+			// We only need to read one line
+			line := in.Text()
+			log.Debugf("tegrastats: %s", line)
+			if err = c.processTegraStatsOutput(line); err != nil {
+				_ = log.Error(err)
+			}
+		} else {
+			_ = log.Warnf("tegrastats did not produce any output")
+		}
+		err = cmd.Process.Signal(os.Kill)
```

Comment:
Is tegrastats running for ever ? Do we need to kill it ?

Why not use  `exec.CommandContext` instead ?

Reply:
Could we then add a comment explaining why we have to kill it then ?